internal/restinterface/internalhandler/internalhandler.go

```diff
@@ -258,17 +260,10 @@ func (h *Handler) APIV1ScoringmgrScoreLibnameGet(w http.ResponseWriter, r *http.
 		return
 	}
 
-	encryptBytes, _ := ioutil.ReadAll(r.Body)
-	Info, err := h.Key.DecryptByteToJSON(encryptBytes)
```

Comment:
Please take into account encoding and decoding like this line.

Reply:
There is no request body for decryption and hence the lines are removed

Reply:
My intention is that `devID` of the request would be encrypted and `APIV1ScoringmgrScoreLibnameGet` should decrypt it to know the actual `devID`.

Reply:
Since this is API and shall be used by other applications to get score of HE, so thought of allowing the apps to directly send the devID in the URL and hence no ecrytpion was used. But if encrypted ID is expected to be sent from third party app, we should provide API to encrypt the simple ID sent and return to the apps which shall be used in this API
please suggest on this

Reply:
This is the internal REST API that only HE uses, not third party app. 
If you run HE with secure mode or enabling sha256 cipher, you can find the devID is encrypted.

I edited the code like below for the test.
**git diff**
```
diff --git a/cmd/edge-orchestration/main.go b/cmd/edge-orchestration/main.go
index 55f19e3..368600f 100644
--- a/cmd/edge-orchestration/main.go
+++ b/cmd/edge-orchestration/main.go
@@ -104,7 +104,7 @@ func orchestrationInit() error {
                }
        }

-       cipher := dummy.GetCipher(cipherKeyFilePath)
+       cipher := sha256.GetCipher(cipherKeyFilePath)
        if isSecured {
                cipher = sha256.GetCipher(cipherKeyFilePath)
        }
diff --git a/internal/restinterface/internalhandler/internalhandler.go b/internal/restinterface/internalhandler/internalhandler.go
index 71e908e..4ad1f31 100644
--- a/internal/restinterface/internalhandler/internalhandler.go
+++ b/internal/restinterface/internalhandler/internalhandler.go
@@ -259,6 +259,7 @@ func (h *Handler) APIV1ScoringmgrScoreLibnameGet(w http.ResponseWriter, r *http.
        }

        encryptBytes, _ := ioutil.ReadAll(r.Body)
+       log.Info(logPrefix, string(encryptBytes))
        Info, err := h.Key.DecryptByteToJSON(encryptBytes)
        if err != nil {
                log.Error(logPrefix, cannotDecryption, err.Error())
```

**dummy cipher log**
```
INFO[2022-08-23T05:41:54Z]internalhandler.go:262 APIV1ScoringmgrScoreLibnameGet [RestInternalInterface]{"devID":"edge-orchestration-7dfb7aaa-1d72-42ca-a900-2c40e9c0c594"}
INFO[2022-08-23T05:41:54Z]route.go:132 func1 From [10.113.171.82:47234] GET /api/v1/scoringmgr/score APIV1ScoringmgrScoreLibnameGet 1.29907ms
```

**sha256 cipher log**
```
INFO[2022-08-23T05:38:21Z]internalhandler.go:262 APIV1ScoringmgrScoreLibnameGet [RestInternalInterface]�8ǒ��'gO%C��F�[�
��7��mI��;��>���凝�(���Ec�tO��(6p~��D�����rl6   �[�����Y[r�Ǣ^��¡Q
INFO[2022-08-23T05:38:21Z]route.go:132 func1 From [10.113.171.82:45534] GET /api/v1/scoringmgr/score APIV1ScoringmgrScoreLibnameGet 1.020938ms
```

Reply:
An easy alternative is that we can change this to POST. Which one do you prefer?